src/libutil/source-accessor.hh

```diff
@@ -176,6 +176,67 @@ struct SourceAccessor : std::enable_shared_from_this<SourceAccessor>
         const CanonPath & path,
         SymlinkResolution mode = SymlinkResolution::Full);
 
+    /**
+     * When `resolveSymlinks` encounters an absolute target, that path
+     * may or may not denote a location inside the current source tree
+     * --- exactly when this happens depends on the specific source
+     *  accessor in question, and what tree it "represents".
+     *
+     * For now, we use this variable to decide: if the symlink has this
+     * path as a prefix, then the suffix points inside this source
+     * accessor. If the path does not have path as prefix, than the
+     * symlink escapes the source acccessor.
+     *
+     * Here are three examples:
+     *
+     * - ```
+     *   sourceAccessor.ownLocationForSymlinkResolution = "C:\Foo";
+     *   symlinkTarget = "C:\Foo\baz";
+     *   ```
+     *   Answer: resides inside source `sourceAccessor`
+     *
+     * - ```
+     *   sourceAccessor.ownLocationForSymlinkResolution = "C:\Foo";
+     *   symlinkTarget = "C:\bar\baz";
+     *   ```
+     *   Answer: does not reside inside source `sourceAccessor`
+     *
+     * - ```
+     *   sourceAccessor.ownLocationForSymlinkResolution = "C:\Foo";
+     *   symlinkTarget = "..\..\..";
+     *   ```
+     *   Answer: depends on whether the symlink is deep enough inside
+     *   the source tree
+     *
+     * (I made these examples use Windows/DOS paths to remind the reader
+     * that neither the symlink target nor
+     * `ownLocationForSymlinkResolution` should be a `CanonPath`.)
```

Comment:
I'm not sure how this would work since our source accessor abstraction has no concept of drive letters. Its concept of a path *is* the `CanonPath`.